.golangci.yml

```diff
@@ -0,0 +1,18 @@
+# See https://golangci-lint.run/usage/configuration/#config-file for configuration options.
+
+linters-settings:
+  golint:
+    min-confidence: 0.8
+  gocyclo:
+    min-complexity: 50
```

Comment:
BTW: setting complexity to 50 you can as well take the linter out. If you don't want to change some of the complex functions or ignore some linter warnings that are not really applicable, you can always put a comment in the code above (or behind) the line that a specific linter complains about:

```go
// notlint: gocyclo
func someVeryComplexFunc() {}
```

Reply:
There even is a linter to complain about such comments that are not needed any more, in case that becomes a problem some day.

Reply:
> BTW: setting complexity to 50 you can as well take the linter out. Default is already 30 (which is high!). Recommendation is between 10-20.

We had some complex methods when this was added. We did not want to ignore them - instead we wanted to fix them over time.
So we will do so. Each time we make the complexity lower we can lower this value.

Reply:
well... now you are ignoring all functions. Adding a comment to ignore them and a TODO or Issue would better, I think.

Reply:
> now you are ignoring all functions

I don't understand your point. We are slowly reducing complexity.
If we support writing exclusions from these checks then they diminish the requirement.

Reply:
Too much discussion. Moving on.